osu.Game/Screens/Edit/Editor.cs

```diff
@@ -841,7 +842,26 @@ private EditorMenuItem createDifficultyCreationMenu()
         }
 
         protected void CreateNewDifficulty(RulesetInfo rulesetInfo)
-            => loader?.ScheduleSwitchToNewDifficulty(editorBeatmap.BeatmapInfo.BeatmapSet, rulesetInfo, GetState());
+        {
+            if (!rulesetInfo.Equals(editorBeatmap.BeatmapInfo.Ruleset))
+            {
+                switchToNewDifficulty(rulesetInfo, true);
+                return;
+            }
+
+            dialogOverlay.Push(new CreateNewDifficultyDialog(clearAllObjects => switchToNewDifficulty(rulesetInfo, clearAllObjects)));
```

Comment:
Note if that the user has unsaved changes in the current edited map and requests a new difficulty creation, this creation dialog will show first, followed by the one asking to confirm the difficulty save. Unsure if this is correct, I can make the save implicit (i.e. requesting a new difficulty to be created automatically saves all pending changes).

Reply:
I think that sounds correct. Surprised it actually works as expected though.